wasm/src/diagnostic.rs

```diff
@@ -91,17 +88,14 @@ impl VSDiagnostic {
     /// Any labels with associated messages are included as "related information"
     /// objects in the [VSDiagnostic], whether they fall in the current document or not.
     /// Editors can display these as links to other source locations.
-    fn new<T>(labels: Vec<Label>, source_name: &str, err: &T) -> VSDiagnostic
```

Comment:
Why remove `source_name`? It was being used to get the correct offsets for the source

Reply:
If the `label` is coming from a file that is _not_ equal to `source_name` (as is the case with the broken test below), then the offsets should apply to _that_ file, not the one indicated by `source_name`. That's what this line was ensuring